lib/ssz_ex.ex

```diff
@@ -114,6 +145,46 @@ defmodule LambdaEthereumConsensus.SszEx do
     end
   end
 
+  defp decode_bitlist(bit_list, max_size) do
+    num_bytes = byte_size(bit_list)
+    num_bits = num_bytes * @bits_per_byte
+    padding = num_bits - bit_size(bit_list)
+    formatted_bit_list = <<0::size(padding), bit_list::bitstring>>
+    len = length_of_bitlist(formatted_bit_list)
+
+    cond do
+      len < 0 ->
+        {:error, "missing length information"}
+
+      div(len, @bits_per_byte) + 1 != num_bytes ->
+        {:error, "invalid byte count"}
+
+      len > max_size ->
+        {:error, "out of bounds"}
+
+      true ->
+        pre_trailing_bits_size = (num_bytes - 1) * @bits_per_byte
+        <<pre::size(pre_trailing_bits_size), last_byte::binary>> = formatted_bit_list
+        <<_trailing::1, last::size(7)>> = last_byte
+
+        cond do
+          last == 0 ->
+            {:ok, pre}
+
+          pre == 0 ->
+            {:ok, <<last::size(len)>>}
+
+          true ->
+            {:ok, <<pre::size(pre_trailing_bits_size), last::size(len - pre_trailing_bits_size)>>}
```

Comment:
This is still overly complicated and doesn't address the padding problem. We can simplify this to something shorter:

Again, let's address this. It can be done in a simpler way:

```elixir
size = byte_size(bitlist)
<<pre::size(byte_size-1), last_byte::1>> = bitlist
decoded = <<pre, remove_trailing_bit(last_byte)>>
cond do
  decoded == <<>> -> {:error, "missing length information"}
  # Other edge cases
  true -> decoded
end
```

And that's it. The only missing thing is to code the removal of the trailing bit. Instead of using the length_of_bitlist you can just code a new function that's similar to leading_zeros (matching each possible byte) that ignores the first `1` bit and returns the rest. In the case of a `<<0>>` byte you can just return `<<>>` and the rest of the functionality will stil work.